corehq/ex-submodules/casexml/apps/case/xform.py

```diff
@@ -272,10 +272,24 @@ def close_extension_cases(case_db, cases, device_id):
 
 
 def get_all_extensions_to_close(domain, cases):
-    if not EXTENSION_CASES_SYNC_ENABLED.enabled(domain):
+    if not toggles.EXTENSION_CASES_SYNC_ENABLED.enabled(domain):
         return set()
-    case_ids = [case.case_id for case in cases if case.closed]
-    return CaseAccessors(domain).get_extension_chain(case_ids, include_closed=False)
+    if not toggles.USH_DONT_CLOSE_PATIENT_EXTENSIONS.enabled(domain):
+        case_ids = [case.case_id for case in cases if case.closed]
+        return CaseAccessors(domain).get_extension_chain(case_ids, include_closed=False)
+    else:
+        # super-custom one-off excepton for USH projects
+        PATIENT_CASE_TYPE = 'patient'
+        CONTACT_CASE_TYPE = 'contact'
+        patient_case_ids = [case.case_id for case in cases if case.closed and case.type == PATIENT_CASE_TYPE]
+        patient_extensions = CaseAccessors(domain).get_extension_chain(patient_case_ids, include_closed=False)
+        valid_extensions = [
+            case.id
+            for case in CaseAccessors(domain).get_cases(patient_extensions)
```

Comment:
Perhaps it would be possible to modify `get_extension_chain` to filter on case-type as well, but that also involves looking up the case table anyway (via a foreign lookup). Since that kind of filtering is only needed for this custom use case, I kept these as separate queries explicitly.